newIDE/app/src/ObjectsList/index.js

```diff
@@ -809,90 +1331,147 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
         ].filter(Boolean);
       },
       [
-        changeObjectTags,
-        copyObject,
-        cutObject,
-        deleteObject,
+        copyObjectFolderOrObjectWithContext,
+        cutObjectFolderOrObjectWithContext,
+        deleteObjectFolderOrObjectWithContext,
         duplicateObject,
         editName,
-        getAllObjectTags,
         onAddNewObject,
         onAddObjectInstance,
         onEditObject,
         onExportObject,
         onSelectAllInstancesOfObjectInLayout,
-        openEditTagDialog,
         paste,
         project,
         setAsGlobalObject,
         eventsFunctionsExtensionWriter,
         preferences.values.userShortcutMap,
         canSetAsGlobalObject,
         initialInstances,
+        selectObjectFolderOrObjectWithContext,
+        objectsContainer,
+        onObjectModified,
       ]
     );
 
     // Force List component to be mounted again if project or objectsContainer
     // has been changed. Avoid accessing to invalid objects that could
     // crash the app.
     const listKey = project.ptr + ';' + objectsContainer.ptr;
-
-    const screenType = useScreenType();
+    const initiallyOpenedNodeIds = [
+      projectObjectFolderOrObjectsList.length === 0
+        ? null
+        : globalObjectsRootFolderId,
+      sceneObjectsRootFolderId,
+    ].filter(Boolean);
+
+    const arrowKeyNavigationProps = React.useMemo(
+      () => ({
+        onArrowRight: (item, isOpened) => {
+          if (item.isPlaceholder || item.isRoot) return null;
+          if (!item.objectFolderOrObject.isFolder()) return null;
+          if (!isOpened) {
+            if (treeViewRef.current) {
+              treeViewRef.current.openItems([getTreeViewItemId(item)]);
+            }
+            return null;
+          } else {
+            if (item.objectFolderOrObject.getChildrenCount() === 0) return null;
+            return {
+              objectFolderOrObject: item.objectFolderOrObject.getChildAt(0),
+              global: item.global,
+            };
+          }
+        },
+        onArrowLeft: (item, isOpened) => {
+          if (item.isPlaceholder || item.isRoot) return null;
+          if (!item.objectFolderOrObject.isFolder() || !isOpened) {
+            const parent = item.objectFolderOrObject.getParent();
+            if (parent.getFolderName() === '__ROOT') return null;
```

Comment:
Would be maybe worth using a method `parent.IsRootFolder()`. The C++ class can tell you, it just has to check if it's a folder that has no parent.

The risk of using __ROOT is that this will break when I will use "__ROOT" as the name, and this is not type safe (this will break when I rework the code and type `_ROOT` instead of `__ROOT`).

Reply:
(this is a way to say "no magic strings" :))